trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/objectstore/ObjectStore.java

```diff
@@ -11,23 +11,13 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.trino.s3.proxy.server.rest;
+package io.trino.s3.proxy.server.objectstore;
 
 import jakarta.ws.rs.core.UriBuilder;
 
 import java.net.URI;
 
-@FunctionalInterface
-public interface S3EndpointBuilder
+public interface ObjectStore
```

Comment:
I'm not sure what to think of the `ObjectStore` name. This interface only really makes sense for S3-like (though not necessarily S3) endpoints. Calling it `ObjectStore` IMHO gives the impression that implementations of this interface deal with the quirks of each storage system, yet they would not.

What about `S3StoreEndpointBuilder`?

Reply:
I'm thinking we'll have more functionality in the future besides building endpoints. What about `RemoteS3Facade` or something like that?

Reply:
Please do

Reply:
#37 - done